src/ensembl/src/content/app/entity-viewer/gene-view/components/proteins-list/proteins-list-item-info/ProteinsListItemInfo.tsx

```diff
@@ -60,54 +61,66 @@ const ProteinsListItemInfo = (props: Props) => {
     transcriptWithProteinDomains,
     setTranscriptWithProteinDomains
   ] = useState<Transcript | null>(null);
-  const [proteinSummary, setProteinSummary] = useState<ProteinSummary | null>(
-    null
-  );
+  const [proteinSummary, setProteinSummary] = useState<
+    ProteinSummary | null | undefined
+  >();
+
+  const proteinId =
+    transcript.product_generating_contexts[0].product.unversioned_stable_id;
 
   useEffect(() => {
     const abortController = new AbortController();
-    const proteinId =
-      transcript.product_generating_contexts[0].product.unversioned_stable_id;
-
-    Promise.all([
-      fetchProteinDomains(proteinId, abortController.signal),
-      fetchProteinSummary(proteinId, abortController.signal)
-    ]).then(([proteinDomains, proteinSummaryData]) => {
-      if (!abortController.signal.aborted) {
-        setTranscriptWithProteinDomains(
-          addProteinDomains(transcript, proteinDomains)
-        );
-        proteinSummaryData && setProteinSummary(proteinSummaryData);
+    fetchProteinDomains(proteinId, abortController.signal).then(
+      (proteinDomains) => {
+        if (!abortController.signal.aborted) {
+          setTranscriptWithProteinDomains(
+            addProteinDomains(transcript, proteinDomains)
+          );
+        }
+      }
+    );
+
+    fetchProteinSummary(proteinId, abortController.signal).then(
+      (proteinSummaryData) => {
+        if (!abortController.signal.aborted) {
+          proteinSummaryData
+            ? setProteinSummary(proteinSummaryData)
+            : setProteinSummary(null);
+        }
       }
-    });
+    );
 
     return function cleanup() {
       abortController.abort();
     };
   }, [transcript.stable_id]);
 
-  if (!transcriptWithProteinDomains) {
-    return null;
-  }
-
-  const {
-    product
-  } = transcriptWithProteinDomains.product_generating_contexts[0];
+  const { product } =
+    transcriptWithProteinDomains?.product_generating_contexts[0] || {};
 
   // FIXME: the 695 below is by now a very magic number (also exists in transcript images);
   // we need to move it to a constant
   return (
     <div className={styles.proteinsListItemInfo}>
-      <>
-        <ProteinDomainImage
-          proteinDomains={product.protein_domains}
-          trackLength={trackLength}
-          width={695}
-        />
-        <ProteinImage product={product} trackLength={trackLength} width={695} />
-        <div className={styles.proteinSummary}>
-          <div className={styles.proteinSummaryTop}>
-            {proteinSummary && (
+      {product && (
+        <>
+          <ProteinDomainImage
+            proteinDomains={product.protein_domains}
+            trackLength={trackLength}
+            width={695}
+          />
+          <ProteinImage
+            product={product}
+            trackLength={trackLength}
+            width={695}
```

Comment:
As a future todo: we have a lot of those magic `695`s floating around, both in typescript files and in scss files. Would be great if we could just use a single constant importable both in the script and the style files, like we did with colours for our palette.

Doesn't block this PR, but we should clean this up.

Reply:
Created a ticket to clean it up:
https://www.ebi.ac.uk/panda/jira/browse/ENSWBSITES-800